Objects/bytes_methods.c

```diff
@@ -557,10 +557,49 @@ find_internal(const char *str, Py_ssize_t len,
 }
 
 PyObject *
-_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *sub,
+_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *subobj,
                Py_ssize_t start, Py_ssize_t end)
 {
-    Py_ssize_t result = find_internal(str, len, "find", sub, start, end, +1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        ADJUST_INDICES(start, end, len);
+        // Work in batches of 10000
+        for (; result == -1 && start <= end; start += 10000) {
+            for (Py_ssize_t i = 0; i < PyTuple_GET_SIZE(subobj); i++) {
+                PyObject *subseq = PyTuple_GET_ITEM(subobj, i);
+                Py_ssize_t sublen;
+                Py_buffer subbuf;
+                if (!PyObject_CheckBuffer(subseq)) {
+                    sublen = 1;
+                }
+                else if (PyObject_GetBuffer(subseq, &subbuf,
+                                            PyBUF_SIMPLE) != 0)
+                {
+                    return NULL;
+                }
+                else {
+                    sublen = subbuf.len;
+                }
+                Py_ssize_t cur_end = start + 10000 + sublen;
+                if (cur_end > end) {
+                    cur_end = end;
+                }
+                Py_ssize_t new_result = find_internal(str, len, "find", subseq,
+                                                      start, cur_end, +1);
+                if (new_result == -2) {
+                    return NULL;
+                }
+                if (new_result != -1 &&
+                    (new_result < result || result == -1))
+                {
+                    result = new_result;
+                }
```

Comment:
If `cur_end` is adjusted like the above comment, these lines can be simplified like:

```c
if (new_result != -1) {
    result = new_result;
}
```

Reply:
Currently, if 2 substrings match at the same index, the first one in argument tuple gets priority. This would change priority to the last.

Reply:
that's why I use -1.

Reply:
Oh wait, `.find()` returns just only index. Caller doesn't know which substring is matched.

Reply:
Simplified.